tests/rust/pipe-test/create_pipe/mod.rs

```diff
@@ -50,15 +66,35 @@ fn create_pipe_with_same_name(libos: &mut LibOS, pipe_name: &str) -> Result<()>
         Err(e) => Err(anyhow::anyhow!("create_pipe() failed ({})", e)),
     };
 
-    // Close first pipe.
     match libos.close(pipeqd) {
         Ok(_) => (),
         Err(e) => {
             let errmsg: String = format!("close() failed ({})", e);
             crate::update_error!(ret, errmsg);
             println!("[ERROR] leaking pipeqd={:?}", pipeqd);
         },
-    }
+    };
 
     ret
 }
+
+/// Creates a pipe with a valid name.
+fn create_pipe_with_name(libos: &mut LibOS, pipe_name: &str) -> Result<()> {
+    create_pipe(libos, pipe_name)?;
+    Ok(())
+}
+
+/// Creates a pipe with a valid name and closes it.
+fn create_pipe_with_name_and_close(libos: &mut LibOS, pipe_name: &str) -> Result<()> {
+    let pipeqd: QDesc = create_pipe(libos, pipe_name)?;
+    libos.close(pipeqd)?;
+    Ok(())
+}
+
+/// Helper function that creates a pipe.
+fn create_pipe(libos: &mut LibOS, pipe_name: &str) -> Result<QDesc> {
+    match libos.create_pipe(pipe_name) {
+        Ok(pipeqd) => Ok(pipeqd),
+        Err(e) => anyhow::bail!("create_pipe() failed ({})", e),
+    }
+}
```

Comment:
Couldn't we have just two functions?
- `create_pipe()`, which creates a pipe.
- `create_pipe_and_close()`, which creates and closes a pipe.